blocks/hdr/hdr.js

```diff
@@ -0,0 +1,42 @@
+import { getLibs } from '../../scripts/utils.js';
+
+function getMediaFilename(a) {
+  try {
+    const mediaUrl = new URL(a.href);
+    return mediaUrl.pathname;
+  } catch (e) {
+    console.log('Error parsing media url', e);
+  }
+  return '';
+};
+
+export default async function init(el) {
+  const links = el.querySelectorAll(':scope a');
```

Comment:
If you're not doing a direct descendant, you don't need `:scope`.

Gold plating:

```js
querySelectorAll('a[href*=".jpg"], a[href*=".png"]'); // image formats you want to support.
```